src/mapgen_functions.cpp

```diff
@@ -977,382 +980,319 @@ void mapgen_fungal_flowers(map *m, oter_id, mapgendata dat, int, float)
+            else if (std::equal(std::begin(fourways_neswx), std::end(fourways_neswx), std::begin({0,1,1,1,0,1,1,0})))
+                plaza_dir = 0; // north side of rotary/plaza
+            if (plaza_dir > -1) rot = plaza_dir % 4;
+         case 3: // tee
+            if (!roads_nesw[0]) { rot = 2; break; } // E/S/W, rotate 180 degrees
+            if (!roads_nesw[1]) { rot = 3; break; } // N/S/W, rotate 270 degrees
+            if (!roads_nesw[3]) { rot = 1; break; } // N/E/S, rotate  90 degrees
+            break;                                  // N/E/W, don't rotate
+        case 2: // straight or diagonal
+            if (roads_nesw[1] && roads_nesw[3]) { rot = 1; break; }              // E/W, rotate  90 degrees
+            if (roads_nesw[1] && roads_nesw[2]) { rot = 1; diag = true; break; } // E/S, rotate  90 degrees
+            if (roads_nesw[2] && roads_nesw[3]) { rot = 2; diag = true; break; } // S/W, rotate 180 degrees
+            if (roads_nesw[3] && roads_nesw[0]) { rot = 3; diag = true; break; } // W/N, rotate 270 degrees
+            if (roads_nesw[0] && roads_nesw[1]) {          diag = true; break; } // N/E, don't rotate
+            break;                                                               // N/S, don't rotate
+        case 1: // dead end
+            if (roads_nesw[1]) { rot=1; break; } // E, rotate  90 degrees
+            if (roads_nesw[2]) { rot=2; break; } // S, rotate 180 degrees
+            if (roads_nesw[3]) { rot=3; break; } // W, rotate 270 degrees
+            break;                               // N, don't rotate
+    }
+
+    // rotate the arrays left by rot steps
+    nesw_array_rotate<bool>(sidewalks_neswx,8, rot*2);
+    nesw_array_rotate<bool>(roads_nesw,     4, rot);
+    nesw_array_rotate<int> (curvedir_nesw,  4, rot);
+
+    // now we have only these shapes: '   |   '-   -'-   -|-
+
+    if (diag) { // diagonal roads get drawn differently from all other types
+        // draw sidewalks if a S/SW/W neighbor has_sidewalk
+        if ( sidewalks_neswx[4] || sidewalks_neswx[5] || sidewalks_neswx[6] ) {
+            for (int y=0; y<SEEY*2; y++)
+                for (int x=0; x<SEEX*2; x++)
+                    if ( x>y-4 && (x<4 || y>SEEY*2-5 || y>=x) )
+                        m->ter_set(x, y, t_sidewalk);
+        }
+        // draw diagonal road
+        for (int y=0; y<SEEY*2; y++) {
+            for (int x=0; x<SEEX*2; x++) {
+                if ( x > y && // definitely only draw in the upper right half of the map
+                    (
+                        ( x>3 && y<(SEEY*2-4) ) // middle, for both corners and diagonals
+                        ||
+                        ( x<4 && curvedir_nesw[0]<0 ) // diagonal heading northwest
+                        ||
+                        ( y>(SEEY*2-5) && curvedir_nesw[1]>0 ) // diagonal heading southeast
+                    )
+                ) {
+                    if ( x%4 && (x-y==SEEX-1+(1-(rot/2)) || x-y==SEEX+(1-(rot/2))) ) {
```

Comment:
Could use a comment on what is actually happening here. It draws yellow pavement, but what is that expression doing? Why is `x%4` there and no equivalent for `y`?